### PR TITLE
`Development`: Refactor DataExportCreationService and add Art. 15 GDPR README

### PR DESCRIPTION
**Pull request by tobias-lippert**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

The DataExportCreationService has too many dependencies and is overly complex with > 850 LOC
To improve code quality, this service should be refactored.
Additionally, the data export requires a README, see #6851 

### Description

Split this service up into 5  and moved all related service into their own `dataexport` package.

- `DataExportCreationService`: handles the core logic, relies on the other ones
- `DataExportExerciseCreationService`: creates the data for exercise participations
- `DataExportExamCreationService`creates the data for exam participations
- `DataExportQuizExerciseCreationService`: creates the data for quiz exercises (since the quiz exercise export is rather complex compared to the other exercise types)
-  ` DataExportCommunicationDataService` extracts all the data about posts, messages, reactions a user has made.
Additionally, add a readme file, see #6851 

### Steps for Testing
Code Review for the refactoring:
For the readme:
Prerequisites:

1 User with an email address you have access to
Log in to Artemis
Navigate to Privacy
Navigate to Data Export
Request a data export
Wait until you receive your email
Download the data export
Make sure the zip contains the README.md file

#### Performance Review
- [x] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Manual tests
- [x] Manuel Test 1
- [x] Manuel Test 2

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [DataExportCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportCreationService.html) | 97% | ✅ |
| [DataExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportService.html) | 99% | ✅|
| [DataExportCommunicationDataService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportCommunicationDataService.html) | 99% | ✅ |
| [DataExportCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportCreationService.html) | 97% | ✅ |
| [DataExportExamCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportExamCreationService.html) | 97% | ✅|
| [DataExportExerciseCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportExerciseCreationService.html) | 90% | ✅ |
| [DataExportQuizExerciseCreationService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportQuizExerciseCreationService.html) | 100% | ✅ |
| [DataExportUtil.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.dataexport/DataExportUtil.html) | 100% | ✅ |
| [DataExportScheduleService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.scheduled/DataExportScheduleService.html) | 95% | ✅ |
| [DataExportResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5117-1/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/DataExportResource.html) | 98% | ✅ |

### CODE REVIEW

**Review by MaximilianAnzinger — COMMENTED** [review]
Left some suggestions.
Other than that the code looks fine 👍

File: src/main/java/de/tum/in/www1/artemis/service/dataexport/DataExportQuizExerciseCreationService.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+import de.tum.in.www1.artemis.domain.quiz.*;
```
[the remark this answers is not in the document]
Comment (tobias-lippert) [reply src/main/java/de/tum/in/www1/artemis/service/dataexport/DataExportQuizExerciseCreationService.java:13]: I need ten importss to be able to remove the wildcard import, so * import seems to be correct.

**Review by MaximilianAnzinger — APPROVED** [review]

**Review by JohannesStoehr — APPROVED** [review]
Refactoring looks good to me

**Review by MaximilianAnzinger — APPROVED** [review]
Reapproved after merge

**Review by JohannesStoehr — DISMISSED** [review]
Reapprove after merge

**Review by JohannesStoehr — APPROVED** [review]
Tested on TS1 and found no issues

**Review by MaximilianAnzinger — APPROVED** [review]
Tested on TS1 and found no issues

**Review by MaximilianAnzinger — APPROVED** [review]
Tested on TS1 and found no issues

**Pull request merged**